modules/server/src/network/aggregations/AggregationAccumulator.ts

```diff
@@ -163,34 +155,35 @@ export const resolveAggregation = (aggregations: Aggregations[]): Aggregations =
 	return resolvedAggregation;
 };
 
-const resolveNumericAggregation = (aggregations: NumericAggregations) => {
-	// TODO: implement
-	throw Error('Not implemented');
-};
+const resolveNumericAggregation = (aggregations: NumericAggregationsTuple): NumericAggregations => {
+	const resolvedAggregation = aggregations.reduce((resolvedAggregation, agg) => {
+		// max
+		if (agg.stats.max > resolvedAggregation.stats.max) {
+			resolvedAggregation.stats.max = agg.stats.max;
+		}
+		// min
+		if (agg.stats.min < resolvedAggregation.stats.min) {
+			resolvedAggregation.stats.min = agg.stats.min;
+		}
+		// count
+		resolvedAggregation.stats.count += agg.stats.count;
+		// sum
+		resolvedAggregation.stats.sum += agg.stats.sum;
+		// avg
+		resolvedAggregation.stats.avg = resolvedAggregation.stats.sum / resolvedAggregation.stats.count;
 
-const emptyAggregation: Aggregations = { bucket_count: 0, buckets: [] };
+		return resolvedAggregation;
+	});
+	return resolvedAggregation;
+};
 
 export class AggregationAccumulator {
-	requestedFields: string[];
-	totalAgg: AllAggregations;
-
-	constructor(requestedFieldsMap: RequestedFieldsMap) {
-		const requestedFields = Object.keys(requestedFieldsMap);
-		this.requestedFields = requestedFields;
-		/*
-		 * seed accumulator with the requested field keys
-		 * this will make it easier to add to using key lookup instead of Array.find
-		 */
-		this.totalAgg = requestedFields.reduce<AllAggregations>((accumulator, field) => {
-			return { ...accumulator, [field]: emptyAggregation };
-		}, {});
-	}
```

Comment:
not needed anymore, saves an iteration by checking if aggregation exists while resolving